homeassistant/components/devolo_home_network/binary_sensor.py

```diff
@@ -0,0 +1,118 @@
+"""Platform for binary sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from devolo_plc_api.device import Device
+
+from homeassistant.components.binary_sensor import (
+    DEVICE_CLASS_PLUG,
+    DEVICE_CLASS_UPDATE,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ENTITY_CATEGORY_DIAGNOSTIC
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONNECTED_PLC_DEVICES,
+    CONNECTED_TO_ROUTER,
+    DOMAIN,
+    FIRMWARE_UPDATE_AVAILABLE,
+)
+from .entity import DevoloEntity
+
+
+@dataclass
+class DevoloBinarySensorRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    value_func: Callable[[DevoloBinarySensorEntity], bool]
+
+
+@dataclass
+class DevoloBinarySensorEntityDescription(
+    BinarySensorEntityDescription, DevoloBinarySensorRequiredKeysMixin
+):
+    """Describes devolo sensor entity."""
+
+
+SENSOR_TYPES: dict[str, DevoloBinarySensorEntityDescription] = {
+    CONNECTED_TO_ROUTER: DevoloBinarySensorEntityDescription(
+        key=CONNECTED_TO_ROUTER,
+        device_class=DEVICE_CLASS_PLUG,
+        entity_category=ENTITY_CATEGORY_DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:router-network",
+        name="Connected to router",
+        value_func=lambda entity: all(
+            device["attached_to_router"]
+            for device in entity.coordinator.data["network"]["devices"]
+            if device["mac_address"] == entity._device.mac
+        ),
```

Comment:
Recommended practise is that lambda functions should be one-line only.
If it's multiple lines, create a function and reference it here:
```suggestion
        value_func=_is_connected_to_router
```

Reply:
Makes sense ...